zebra-state/src/service/non_finalized_state.rs

```diff
@@ -418,12 +418,10 @@ impl NonFinalizedState {
         chain_push_result.expect("scope has finished")
     }
 
-    /// Returns the length of the non-finalized portion of the current best chain.
-    pub fn best_chain_len(&self) -> u32 {
-        self.best_chain()
-            .expect("only called after inserting a block")
-            .blocks
-            .len() as u32
+    /// Returns the length of the non-finalized portion of the current best chain
+    /// or `None` if the best chain has no blocks.
+    pub fn best_chain_len(&self) -> Option<u32> {
+        self.best_chain().map(|chain| chain.blocks.len() as u32)
```

Comment:
This is easier to read without `map()`, and it needs a comment explaining why `as` is ok to use:

```suggestion
        // This `as` can't overflow because the number of blocks in the chain is limited to i32::MAX,
        // and the non-finalized chain is further limited by the fork length (slightly over 100 blocks)
        Some(self.best_chain()?.blocks.len() as u32)
```

Reply:
This seems like a bug in the test, so it might have been easier to fix it there.